### PR TITLE
feat(PPDSC-2838): use new s3 for pr links

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2838

- Add new bucket for PR

Please note, we need to update the below in `ncu-newskit-site-pr` context once this ticket is merged.
```
SITE_BASE_URL
http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/
```

Right now clicking on other links are not working. Once the above is done, they will be working again. This has been tested.

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/ppdsc-2838-use-new-s3-for-pr
  Storybook: http://ncu-newskit-docs-pr.s3-website-eu-west-1.amazonaws.com/ppdsc-2838-use-new-s3-for-pr/storybook

File: .circleci/config.yml [hunk 1]
```diff
@@ -1463,11 +1484,11 @@ workflows:
...
             - init_aws
-      - deploy_terraform:
+      - deploy_terraform_oidc:
           name: deploy_terraform_s3_pr
-          package: s3
-          context: ncu-product-platforms-context
-          apply: false
+          package: s3/dev-newskit
+          context: ncu-newskit-site-pr
+          apply: true
```
Comment (Xin00163) [inline .circleci/config.yml:1491]: @JohnTParsons do we need to set it to false now that we finished testing?

Comment (JohnTParsons) [reply .circleci/config.yml:1491]: No, that instance should always be true, as not just for validating TF upfront but for doing it upfront.

File: terraform/s3/dev-newskit/variables.tf [hunk 2]
```diff
@@ -17,5 +17,5 @@ variable "tag_name_suffix" {
 
 variable "docs_bucket" {
   description = "First bucket name per account for the documentation site"
-  default     = "ncu-newskit-docs-pr"
+  default     = "ncu-newskit-docs-pr"
```
Comment (JohnTParsons) [inline terraform/s3/dev-newskit/variables.tf:20]: You can now delete that surplus space you added (to get the TF step to run)

**Review by JohnTParsons — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -1076,22 +1078,41 @@ jobs:
...
+          profile-name: 'NEWSKIT-OIDC-PROFILE'
+          role-session-name: '${CIRCLE_WORKFLOW_ID}-s3-push'
+      - node/install:
+          install-yarn: false
+          node-version: '16.13'
       - run:
           name: Upload PR sites to S3 subfolders
           command: |
-            export s3Path_docs=$(node scripts/s3-pr-branch-naming.js ${CIRCLE_BRANCH} 's3://ncu-newskit-docs/')
+            node --version
```
Comment (JohnTParsons) [inline .circleci/config.yml:1106]: "node --version" line could be removed

**Review by JohnTParsons — APPROVED** [review]

File: site/pages/components/text-area.tsx [hunk 4]
```diff
@@ -370,7 +370,7 @@ const TextAreaComponent = (layoutProps: LayoutProps) => (
             <>
               Text Areas can use{' '}
               <Link
-                href="http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2470-character-count-documentation/components/character-count/"
```
Comment (baburay23) [inline site/pages/components/text-area.tsx:373]: looks good to me just wanted to ask is this a mistake leftover from before? Wondering why it was referencing a particular PR link

**Review by baburay23 — APPROVED** [review]

File: site/pages/components/text-area.tsx (hunk 4 above)
Comment (Xin00163) [reply site/pages/components/text-area.tsx:373]: Yeah I think it's a mistake from before.

**Pull request merged**